File: backend/src/main/java/com/suitcustom/entity/Style.java
package com.suitcustom.entity;

import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 款式实体类
 * 
 * @author suitcustom
 */
@Data
public class Style implements Serializable {

  private static final long serialVersionUID = 1L;

  /**
   * 款式ID
   */
  private Long id;

  /**
   * 款式名称
   */
  private String name;

  /**
   * 款式编码
   */
  private String code;

  /**
   * 款式类型（1-西装上衣，2-西装裤，3-马甲，4-衬衫，5-领带）
   */
  private Integer type;

  /**
   * 款式图片
   */
  private String image;

  /**
   * 款式模型（3D模型文件路径）
   */
  private String model;

  /**
   * 款式价格（元）
   */
  private BigDecimal price;

  /**
   * 款式描述
   */
  private String description;

  /**
   * 状态（0-下架，1-上架）
   */
  private Integer status;

  /**
   * 创建时间
   */
  private LocalDateTime createTime;

  /**
   * 更新时间
   */
  private LocalDateTime updateTime;
}

File: frontend/customer-web/src/api/style.ts
import request from '@/utils/request'
import type { ApiResponse } from '@/types/api'
import type { StyleInfo as StyleInfoType } from '@/types/style'

// 款式信息接口
export interface StyleInfo {
  id: number
  name: string
  code: string
  type: number
  image: string
  model: string
  price: number
  description: string
  status: number
  createTime: string
  updateTime: string
}

// 获取款式列表
export function getStyleList() {
  return request.get<StyleInfoType[]>('/style/list')
}

// 获取款式详情
export function getStyleDetail(id: number) {
  return request.get<StyleInfoType>(`/style/${id}`)
}

// 获取推荐款式
export function getRecommendStyles() {
  return request.get<StyleInfoType[]>('/style/recommend')
}

// 检查款式编码是否存在
export function checkStyleCode(code: string) {
  return request.get<boolean>('/style/check/code', { params: { code } })
} 